pytorch_lightning/utilities/model_summary.py

```diff
@@ -261,7 +261,11 @@ def _forward_example_input(self) -> None:
         input_ = model.example_input_array
         input_ = model._apply_batch_transfer_handler(input_)
 
-        if trainer is not None and trainer.amp_backend == AMPType.NATIVE and trainer._device_type != DeviceType.TPU:
+        if (
+            trainer is not None
+            and trainer.amp_backend == AMPType.NATIVE
+            and trainer._device_type != _AcceleratorType.TPU
```

Comment:
```suggestion
            and not isinstance(trainer.accelerator, TPUAccelerator)
```
Outside of the trainer/accelerator connector, this is how we want to check for the accelerator type in the future. Just a reminder, let's update this in follow up PRs

Reply:
Yup!